authors/apps/articles/views.py

```diff
@@ -116,14 +121,27 @@ def post(self, request, slug):
         """
         Method that posts users article ratings
         """
+        rating = request.data.get("rate", {})
+        serializer = self.serializer_class(data=rating)
+        serializer.is_valid(raise_exception=True)
+        rating = serializer.data.get('rating')
         try:
             article = Article.objects.get(slug=slug)
         except Article.DoesNotExist:
             raise NotFound("An article with this slug does not exist")
-
-        rate = request.data.get('rate', {})
-        serializer = self.serializer_class(article, data=rate, partial=True)
-        serializer.is_valid(raise_exception=True)
-        serializer.save()
-
-        return Response(serializer.data, status=status.HTTP_201_CREATED)
+        ratings = Ratings.objects.filter(rater=request.user.profile, article=article).first()
+        if not ratings:
+            ratings = Ratings(article=article, rater=request.user.profile, stars=rating)
+            ratings.save()
+            avg = Ratings.objects.filter(article=article).aggregate(Avg('stars'))
+            return Response({
+                "avg":avg
+                }, status=status.HTTP_201_CREATED)
+
+        if ratings.counter >= 5: 
+            return Response({"message":"You are not allowed to rate this article more than 5 times."}, status=status.HTTP_403_FORBIDDEN)
```

Comment:
I think you should raise a permission denied exception from rest_framework.exceptions instead of returning a message here.